### PR TITLE
[INLONG-2545][Agent] agent pull task (binlog,file) from manager

### LINKED ISSUE #2545: [Feature]agent pull task (file,binlog ) from manager
### Description

agent pull task (file,binlog ) from manager

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by LvJiancheng**
### Title Name: [INLONG-2545] [Agent]agent pull task (binlog,file) from manager

where *XYZ* should be replaced by the actual issue number.

Fixes #2545

### Motivation

*expend tasktype of task from manager in agent *

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/enums/ManagerOpEnum.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.agent.enums;
+
+import static java.util.Objects.requireNonNull;
+
+public enum ManagerOpEnum {
```
Comment (EMsnap) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/enums/ManagerOpEnum.java:22]: PLS extract to inlong common

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/enums/TaskTypeEnum.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.agent.enums;
+
+import static java.util.Objects.requireNonNull;
+
+public enum TaskTypeEnum {
```
Comment (EMsnap) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/enums/TaskTypeEnum.java:22]: Pls extract to inlong common

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/JobProfileDto.java [hunk 3]
```diff
@@ -91,27 +144,63 @@
...
+
+        binlogJob.setTrigger(DEFAULT_TRIGGER);
+        binlogJob.setChannel(DEFAULT_CHANNEL);
+        binlogJob.setName(MANAGER_JOB);
+        binlogJob.setSource(DEFAULT_SOURCE);
+        binlogJob.setSink(DEFAULT_DATAPROXY_SINK);
+        binlogJob.setOp(dataConfigs.getOp());
+        binlogJob.setDeliveryTime(dataConfigs.getDeliveryTime());
+
+        binlogJob.setJob_database_hostname(binlogJobTaskConfig.getJob_database_hostname());
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/JobProfileDto.java]: Camel-Case pls

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/JobProfileDto.java [hunk 4]
```diff
@@ -58,24 +62,73 @@
...
+
+        private String op;
+        private String trigger;
+        private String name;
+        private String source;
+        private String sink;
+        private String channel;
+        private String deliveryTime;
+
+        private  String job_database_user;
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/JobProfileDto.java]: camel case

**Review by EMsnap — COMMENTED** [review]
pls take a look

**Review by binaryWorld — COMMENTED** [review]
TaskRequestDto、TaskResult  advised to  inlong-common  module

File: inlong-common/pom.xml [hunk 5]
```diff
@@ -63,11 +63,30 @@
         <dependency>
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
-            <version>1.18.12</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.sleepycat</groupId>
+            <artifactId>je</artifactId>
+            <version>7.3.7</version>
```
Comment (aloyszhang) [inline inlong-common/pom.xml:71]: version for dependency should be specified in `properties`

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/BinlogJob.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.agent.plugin.fetcher.dtos;
+
+import lombok.Data;
+
+@Data
+public class BinlogJob {
```
Comment (aloyszhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/BinlogJob.java:23]: Why not abstract an interface for Job?

**Review by aloyszhang — COMMENTED** [review]
Thanks for your contribution, left some comments, PTAL.

**Review by EMsnap — CHANGES_REQUESTED** [review]
PLAL

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/JobProfileDto.java [hunk 7]
```diff
@@ -130,11 +219,21 @@ public static TriggerProfile convertToTriggerProfile(DataConfig dataConfigs) {
...
         JobProfileDto profileDto = new JobProfileDto();
         Proxy proxy = getProxy(dataConfigs);
-        Job job = getJob(dataConfigs);
+        switch (taskType) {
+            case SQL:
+            case BINLOG:
+                BinlogJob binlogJob = getBinlogJob(dataConfigs);
+                profileDto.setBinlogJob(binlogJob);
+            case FILE:
+                FileJob job = getFileJob(dataConfigs);
```
[the remark this answers is not in the document]
Comment (LvJiancheng) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/JobProfileDto.java]: done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/JobProfileDto.java (hunk 4 above)
Comment (LvJiancheng) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/JobProfileDto.java]: done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/FileJob.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    @Data
+    public static class Thread {
+
+        private Running running;
+    }
+
+    @Data
+    public static class FileJobTaskConfig {
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/FileJob.java:62]: when change to FileJob, the properties in JobConstants should be job.filejob.file.read.wait

**Review by EMsnap — COMMENTED** [review]
PLAL

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/BinlogJob.java (hunk 6 above)
Comment (LvJiancheng) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/BinlogJob.java:23]: good idea,fix it later

**Review by EMsnap — APPROVED** [review]
+1

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/BinlogJob.java (hunk 6 above)
Comment (LvJiancheng) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/dtos/BinlogJob.java]: done

**Review by dockerzhang — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]
+1

**Pull request merged**